### ISSUE TITLE
Use of pango_font_face_is_synthesized (was: 'Minimum Pango version)

### ISSUE THREAD

**Issue by amitdo** [tesseract-ocr/tesseract#423]
Tesseract training tool text2image uses these two functions: 
[pango_glyph_item_iter_init_start](https://github.com/tesseract-ocr/tesseract/blob/193032a7786f/training/pango_font_info.cpp#L455)
[pango_glyph_item_iter_next_cluster](https://github.com/tesseract-ocr/tesseract/blob/193032a7786f/training/pango_font_info.cpp#L458)

 That means it requires Pango >=1.22.0:
https://developer.gnome.org/pango/stable/api-index-1-22.html
https://developer.gnome.org/pango/stable/pango-Glyph-Storage.html#pango-glyph-item-iter-init-start
https://developer.gnome.org/pango/stable/pango-Glyph-Storage.html#pango-glyph-item-iter-next-cluster

However, the comments and code in the method [`bool FontUtils::IsAvailableFont()`](https://github.com/tesseract-ocr/tesseract/blob/193032a7786f/training/pango_font_info.cpp#L509) in `/training/pango_font_info.cpp` refer to older Pango versions:

> // The generated list of font families and faces includes "synthesized" font
> // faces that are not truly loadable. **Pango versions >=1.18** have a
> // pango_font_face_is_synthesized method that can be used to prune the list.
> // **Until then**, we are restricted to using a hack...

`if (PANGO_VERSION <= 12005) {`

`12005` means 1.20.5
https://github.com/GNOME/pango/blob/1147da131ad13/pango/pango-utils.h#L136

**Comment by jollysean** [comment]
I am facing issues in window installation. And probably due to Pango / Cairo. Any help please?
c:\work\tesseract\build>cmake .. -DSTATIC=1
-- Downloading latest ICU binaries
-- Checking for module 'pango'
--
CMake Error at C:/Program Files/CMake/share/cmake-3.7/Modules/FindPkgConfig.cmak
e:415 (message):
  A required package was not found
```
Call Stack (most recent call first):
  C:/Program Files/CMake/share/cmake-3.7/Modules/FindPkgConfig.cmake:588 (_pkg_c
heck_modules_internal)
  training/CMakeLists.txt:190 (pkg_check_modules)


-- Configuring incomplete, errors occurred!
See also "C:/work/tesseract/build/CMakeFiles/CMakeOutput.log".
See also "C:/work/tesseract/build/CMakeFiles/CMakeError.log".
```

**Comment by Brian51** [comment]
http://www.pango.org/Download

**Comment by jollysean** [comment]
404 error https://www.gtk.org/download/win64.php
Tried [this](https://sourceforge.net/projects/sdlpango/)  but no help

**Comment by Shreeshrii** [comment]
@amitdo Can this be closed?

**Comment by amitdo** [comment]
No, although a PR I sent did something to improve it, there are still a place to further improvement.

**Comment by zdenop** [comment]
@Shreeshrii : `PANGO_VERSION <= 12005` condition(1.20.5) is used only in unittest/pango_font_info_test.cc. Because [required version for build 1.22.0](https://github.com/tesseract-ocr/tesseract/blob/1ec34378d9c8a78c46e94c2a441ad8e5c6409141/configure.ac#L463), that part of code is useless.

**Comment by zdenop** [comment]
@amitdo : my test shows that [`FontUtils::IsAvailableFont`](https://github.com/tesseract-ocr/tesseract/blob/1ec34378d9c8a78c46e94c2a441ad8e5c6409141/src/training/pango_font_info.cpp#L493) eliminates more that synthesized fonts (they are not very common, but they were installed by default in my linux distribution). I have no time to make deeper investigation about this, but it seems like `pango_font_face_is_synthesized` is not sufficient for testing font usability for text rendering. So please send PR or please be more specific what should be changed or how `pango_font_face_is_synthesized` should be used. Otherwise please close the issue.

**Comment by amitdo** [comment]
Fixed in #1486 #1487 and commit 1ec34378d9.
